examples/chef/common/chef-fan-control-manager.cpp

```diff
@@ -99,98 +113,205 @@ Status ChefFanControlManager::HandleStep(StepDirectionEnum aDirection, bool aWra
 {
-    VerifyOrDie(aPath.mClusterId == FanControl::Id);
-    VerifyOrDie(aPath.mEndpointId == mEndpoint);
+    ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange");
+    switch (attributeId)
+    {
+    case FanControl::Attributes::PercentSetting::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  PercentSetting");
+        DataModel::Nullable<Percent> percentSetting =
+            NumericAttributeTraits<DataModel::Nullable<Percent>>::StorageToWorking(*value);
+
+        // If the PercentSetting was set to Null (likely the mode was set to kAuto).
+        // Update PercentCurrent to 0 here as it can't be nullable and we don't have a "real"
+        // value for the Fan percentage. The spec doesn't specify what to do in this case.
+        // If not Null, PercentCurrent tracks PercentSetting's value.
+        SetPercentCurrent(percentSetting.IsNull() ? 0 : percentSetting.Value());
+        break;
+    }
+
+    case FanControl::Attributes::SpeedSetting::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  SpeedSetting");
+
+        // DataModel::Nullable<uint8_t> speedSetting = static_cast<DataModel::Nullable<uint8_t>>(*value);
+        DataModel::Nullable<uint8_t> speedSetting = NumericAttributeTraits<DataModel::Nullable<uint8_t>>::StorageToWorking(*value);
+        SetSpeedCurrent(speedSetting.IsNull() ? 0 : speedSetting.Value());
+        // Determine if the speed change should also change the fan mode
+        FanControl::Attributes::FanMode::Set(mEndpoint, SpeedToFanMode(mSpeedCurrent));
+        break;
+    }
+
+    case FanControl::Attributes::FanMode::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  FanMode");
+
+        FanControl::FanModeEnum fanMode = static_cast<FanControl::FanModeEnum>(*value);
+        FanModeWriteCallback(fanMode);
+        break;
+    }
+
+    default: {
+        break;
+    }
+    }
+}
 
-    switch (aPath.mAttributeId)
+FanControl::FanModeEnum ChefFanControlManager::SpeedToFanMode(uint8_t speed)
+{
+    if (speed == 0)
+    {
+        return FanControl::FanModeEnum::kOff;
+    }
+    else if (speed <= FAN_MODE_LOW_UPPER_BOUND)
+    {
+        return FanControl::FanModeEnum::kLow;
+    }
+    else if (speed <= FAN_MODE_MEDIUM_UPPER_BOUND)
+    {
+        return FanControl::FanModeEnum::kMedium;
+    }
+    else if (speed <= FAN_MODE_HIGH_UPPER_BOUND)
     {
-    case SpeedSetting::Id: {
-        Nullable<uint8_t> newSpeedSetting;
-        ReturnErrorOnFailure(aDecoder.Decode(newSpeedSetting));
+        return FanControl::FanModeEnum::kHigh;
+    }
+    return FanControl::FanModeEnum::kUnknownEnumValue;
+}
 
-        // Ensure new speed is in bounds
+void ChefFanControlManager::SetPercentCurrent(uint8_t aNewPercentCurrent)
+{
+    if (aNewPercentCurrent != mPercentCurrent)
```

Comment:
nit: you could decrease indend by doing:

```cpp
if (aNewPercentCurrent != mPercentCurrent)
{
  return;
}
// .. code that is now not indented
```

Reply:
Fixed thanks!